hail/src/main/scala/is/hail/expr/ir/Binds.scala

```diff
@@ -75,70 +90,118 @@ case class SegregatedBindingEnv[A, B](
     copy(newBindings = newBindings.bindRelational(bindings: _*))
 }
 
-case class EvalOnlyBindingEnv[T](env: Env[T]) extends GenericBindingEnv[EvalOnlyBindingEnv[T], T] {
-  override def promoteAgg: EvalOnlyBindingEnv[T] =
-    EvalOnlyBindingEnv(Env.empty)
-
-  override def promoteScan: EvalOnlyBindingEnv[T] =
-    EvalOnlyBindingEnv(Env.empty)
-
-  override def bindEval(bindings: (String, T)*): EvalOnlyBindingEnv[T] =
-    EvalOnlyBindingEnv(env.bindIterable(bindings))
-
-  override def dropEval: EvalOnlyBindingEnv[T] =
-    EvalOnlyBindingEnv(Env.empty)
-
-  override def bindAgg(bindings: (String, T)*): EvalOnlyBindingEnv[T] =
-    this
-
-  override def bindScan(bindings: (String, T)*): EvalOnlyBindingEnv[T] =
-    this
-
-  override def createAgg: EvalOnlyBindingEnv[T] =
-    this
-
-  override def createScan: EvalOnlyBindingEnv[T] =
-    this
-
-  override def noAgg: EvalOnlyBindingEnv[T] =
-    this
-
-  override def noScan: EvalOnlyBindingEnv[T] =
-    this
-
-  override def onlyRelational(keepAggCapabilities: Boolean = false): EvalOnlyBindingEnv[T] =
-    EvalOnlyBindingEnv(Env.empty)
-
-  override def bindRelational(bindings: (String, T)*): EvalOnlyBindingEnv[T] =
-    this
-}
-
 object Binds {
   def apply(x: IR, v: String, i: Int): Boolean =
-    Bindings(x, i, EvalOnlyBindingEnv(Env.empty[Type])).env.contains(v)
+    Bindings.get(x, i, BindingEnv.empty[Type].createAgg.createScan).eval.contains(v)
 }
 
+final case class Bindings(
+  eval: Seq[(String, Type)] = Seq.empty,
```

Comment:
We've adopted an idiom of requiring `IndexedSeq` instead of `Seq` to prevents us using linked lists which have dreadful iteration performance. If there's no reason to use `List`, maybe continue with this trend?

Reply:
`Seq.empty` returns a `mutable.ListBuilder` whereas `IndexedSeq` defaults to `Vector`